msalautomationapp/src/main/java/com/microsoft/identity/client/msal/automationapp/sdk/MsalSdk.java

```diff
@@ -105,12 +108,24 @@ public MsalAuthResult acquireTokenSilent(@NonNull MsalAuthTestParams authTestPar
 
         final ResultFuture<IAuthenticationResult, Exception> future = new ResultFuture<>();
 
-        final IAccount account = getAccount(
+        IAccount account = getAccount(
                 authTestParams.getActivity(),
                 authTestParams.getMsalConfigResourceId(),
                 authTestParams.getLoginHint()
         );
 
+        // Handle accounts for AAD B2C
+        Authority authority = null;
+        try {
+            authority = Authority.getAuthorityFromAuthorityUrl(authTestParams.getAuthority());
+        } catch (IllegalArgumentException e) {
+            authority = pca.getConfiguration().getDefaultAuthority();
+        }
+        if(authority instanceof AzureActiveDirectoryB2CAuthority){
+            final String policyName = ((AzureActiveDirectoryB2CAuthority) authority).getB2CPolicyName();
+            account = getAccountForPolicyName((MultipleAccountPublicClientApplication) pca, policyName);
+        }
+
```

Comment:
I personally don't like this code...we may want to improve this as follows:

```java
final Authority authority;

if (authTestParams.getAuthority() != null) {
   authority = Authority.getAuthorityFromAuthorityUrl(authTestParams.getAuthority());
} else {
   authority = pca.getConfiguration().getDefaultAuthority();
}

final IAccount account;

if (authority instanceof AzureActiveDirectoryB2CAuthority) {
   final String policyName = ((AzureActiveDirectoryB2CAuthority) authority).getB2CPolicyName();
   account = getAccountForPolicyName((MultipleAccountPublicClientApplication) pca, policyName);
} else {
   account = getAccount(
      authTestParams.getActivity(),
      authTestParams.getMsalConfigResourceId(),
      authTestParams.getLoginHint()
   );
}
```

Reply:
Solve here
c4f4affbab7bbf45d59c1f18fbf89de7db4f2add

Reply:
c4f4affbab7bbf45d59c1f18fbf89de7db4f2add